src/parquet/file/reader-internal.cc

```diff
@@ -293,10 +255,13 @@ void SerializedFile::ParseMetaData() {
   if (bytes_read != metadata_len) {
     throw ParquetException("Invalid parquet file. Could not read metadata bytes.");
   }
-  DeserializeThriftMsg(&metadata_buffer[0], &metadata_len, &metadata_);
 
-  schema::FlatSchemaConverter converter(&metadata_.schema[0], metadata_.schema.size());
-  schema_.Init(converter.Convert());
+  auto metadata_ptr = std::unique_ptr<format::FileMetaData>(new format::FileMetaData());
+  metadata_ = metadata_ptr.get();
+  DeserializeThriftMsg(&metadata_buffer[0], &metadata_len, metadata_);
+
+  file_meta_data_ =
+      FileMetaData::GetFileMetaData(reinterpret_cast<uint8_t*>(metadata_ptr.release()));
```

Comment:
Do you have a better solution to this ? The main issue is we can't expose `format::FileMetaData` in the API.
`std::move()` is not atomic too right, won't we hit the same leak issue ?